tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -93,13 +118,86 @@ public async Task<long> UploadRepositoryChangesAsync()
         if (localCommits.Length == 0)
         {
+            Log.Debug("ITR: Local commits not found. (since 1 month ago)");
             return 0;
         }
 
+        Log.Debug<int>("ITR: Local commits = {count}", localCommits.Length);
         var remoteCommitsData = await SearchCommitAsync(localCommits).ConfigureAwait(false);
         return await SendObjectsPackFileAsync(localCommits[0], remoteCommitsData).ConfigureAwait(false);
     }
 
+    public async Task<SkippableTest[]> GetSkippableTestsAsync()
+    {
+        Log.Debug("ITR: Getting skippable tests...");
+        var framework = FrameworkDescription.Instance;
+        var repository = await _getRepositoryUrlTask.ConfigureAwait(false);
+        var currentSha = await ProcessHelpers.RunCommandAsync(new ProcessHelpers.Command("git", "rev-parse HEAD", _workingDirectory)).ConfigureAwait(false);
+        if (currentSha is null)
+        {
+            Log.Warning("ITR: 'git rev-parse HEAD' command is null");
+            return Array.Empty<SkippableTest>();
+        }
+
+        currentSha = currentSha.Replace("\n", string.Empty);
+        var query = new DataEnvelope<Data<SkippableTestsQuery>>(
+            new Data<SkippableTestsQuery>(
+                default,
+                TestParamsType,
+                new SkippableTestsQuery(
+                    repository,
+                    currentSha,
+                    new SkippableTestsConfigurations(
+                        framework.OSPlatform,
+                        Environment.OSVersion.VersionString,
+                        framework.OSArchitecture,
+                        framework.Name,
+                        framework.ProductVersion,
+                        framework.ProcessArchitecture))),
+            default);
+        var jsonQuery = JsonConvert.SerializeObject(query, SerializerSettings);
+        var jsonQueryBytes = Encoding.UTF8.GetBytes(jsonQuery);
+        Log.Debug("ITR: JSON RQ = {json}", jsonQuery);
+
+        return await WithRetries(InternalGetSkippableTestsAsync, jsonQueryBytes, MaxRetries).ConfigureAwait(false);
+
+        async Task<SkippableTest[]> InternalGetSkippableTestsAsync(byte[] state, bool finalTry)
+        {
+            var request = _apiRequestFactory.Create(_skippableTestsUrl);
+            request.AddHeader(ApiKeyHeader, _settings.ApiKey);
+            request.AddHeader(ApplicationKeyHeader, _settings.ApplicationKey);
+            request.AddHeader(HttpHeaderNames.TraceId, _id);
+            request.AddHeader(HttpHeaderNames.ParentId, _id);
+            Log.Debug("ITR: Searching skippable tests from: {url}", _skippableTestsUrl.ToString());
+            var response = await request.PostAsync(new ArraySegment<byte>(state), MimeTypes.Json).ConfigureAwait(false);
+            var responseContent = await response.ReadAsStringAsync().ConfigureAwait(false);
+            if (response.StatusCode is < 200 or >= 300 && response.StatusCode != 404)
```

Comment:
Why ignoring 404s?

Reply:
By `ignoring it` we avoid to retry a 404, if the content is not found retrying it doesn't change that fact.

Reply:
👍 Makes sense. We don't have this in other places, though.